docs/proposals/docs-process.md

```diff
@@ -0,0 +1,94 @@
+
+- Inline/top-of-page Table of Contents can be done with JavaScript, e.g.: http://jsfiddle.net/s8m2t/8/
+- Header/footer insertion can be done by simply using the native layouts functionality in Jekyll
+- Linking to specific versions of docs can be done with inline variables in the URL, e.g. /docs/{{ version }}/hello-world.md
+- Replacement of file extensions in link paths should not be necessary as public Jekyll URLs are "friendly" (extension-free)
+- Other pre/post-processing scripts for docs?
+
+## Adapting the current docs process for releases
+
+Nuts and bolts: How would a pure-GitHub Pages process work? Let’s rewrite this doc that we provide to the community, which covers [how to contribute to our docs for a K8s release](https://github.com/kubernetes/kubernetes/blob/master/docs/devel/update-release-docs.md).
+
+### Adding a new docs collection for a release
+
+Whenever a new release series (release-X.Y) is cut from master, we push the corresponding set of docs to `http://kubernetes.io/vX.Y/docs`. The steps are as follows:
+
+* Create a `_vX.Y` folder in the `kubernetes.github.io` repo.
+
+* Add vX.Y as a valid collection in `_config.yml`.
+
+* Create a new `_includes/nav_vX.Y.html` file with the navigation menu. This can be a copy of `_includes/nav_vX.Y-1.html` with links to new docs added and links to deleted docs removed. Update _layouts/docwithnav.html to include this new navigation html file. Example PR: [#16143](https://github.com/kubernetes/kubernetes/pull/16143).
+
+* [Pull docs from release branch](https://github.com/kubernetes/kubernetes/blob/master/docs/devel/update-release-docs.md#updating-docs-in-gh-pages-branch) in _vX.Y folder.
+
+Once these changes have been submitted, you should be able to reach the docs at http://kubernetes.io/vX.Y/docs/where you can test them.
+
+To make X.Y the default version of docs:
+
+* Update `_config.yml` and `index.md` to point to the new version. Example PR:[#16416](https://github.com/kubernetes/kubernetes/pull/16416).
+
+* Update `_includes/docversionselector.html` to make vX.Y the default version.
+
+* Add `Disallow: /vX.Y-1/` to existing robots.txt file to hide old content from web crawlers and focus SEO on new docs. Example PR: [#16388](https://github.com/kubernetes/kubernetes/pull/16388).
+
+* Regenerate `sitemaps.xml` so that it now contains vX.Y links. Sitemap can be regenerated using [https://www.xml-sitemaps.com](https://www.xml-sitemaps.com/). Example PR: [#17126](https://github.com/kubernetes/kubernetes/pull/17126).
+
+* Resubmit the updated sitemaps file to [Google webmasters](https://www.google.com/webmasters/tools/sitemap-list?siteUrl=http://kubernetes.io/) for google to index the new links.
+
+* Update `_layouts/docwithnav.html` to include `_includes/archivedocnotice.html` for vX.Y-1 docs which need to be archived.
+
+* Ping @johndmulhausen to update docs.k8s.io to redirect to http://kubernetes.io/vX.Y/. [#18788](https://github.com/kubernetes/kubernetes/issues/18788).
```

Comment:
https://jekyllrb.com/docs/plugins/

On Mon, Feb 8, 2016 at 10:37 PM, Eric Tune <etune@google.com> wrote:

> Brian: github does not allow Jekyll plugins.
>
> On Mon, Feb 8, 2016 at 10:09 PM, Brian Grant <notifications@github.com>
> wrote:
>
>> In docs/proposals/docs-process.md
>> <https://github.com/kubernetes/kubernetes/pull/20874#discussion_r52270792>
>> :
>>
>> > +
>> > +To make X.Y the default version of docs:
>> > +
>> > +* Update `_config.yml` and `index.md` to point to the new version. Example PR:[#16416](https://github.com/kubernetes/kubernetes/pull/16416).
>> > +
>> > +* Update `_includes/docversionselector.html` to make vX.Y the default version.
>> > +
>> > +* Add `Disallow: /vX.Y-1/` to existing robots.txt file to hide old content from web crawlers and focus SEO on new docs. Example PR: [#16388](https://github.com/kubernetes/kubernetes/pull/16388).
>> > +
>> > +* Regenerate `sitemaps.xml` so that it now contains vX.Y links. Sitemap can be regenerated using [https://www.xml-sitemaps.com](https://www.xml-sitemaps.com/). Example PR: [#17126](https://github.com/kubernetes/kubernetes/pull/17126).
>> > +
>> > +* Resubmit the updated sitemaps file to [Google webmasters](https://www.google.com/webmasters/tools/sitemap-list?siteUrl=http://kubernetes.io/) for google to index the new links.
>> > +
>> > +* Update `_layouts/docwithnav.html` to include `_includes/archivedocnotice.html` for vX.Y-1 docs which need to be archived.
>> > +
>> > +* Ping @johndmulhausen to update docs.k8s.io to redirect to http://kubernetes.io/vX.Y/. [#18788](https://github.com/kubernetes/kubernetes/issues/18788).
>>
>> We should create a github team alias for that.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/kubernetes/kubernetes/pull/20874/files#r52270792>.
>>
>
>